### ISSUE TITLE
Court User: Ability to put a “hold” on a case

### ISSUE THREAD

**Issue by mariahkannenberg** [ustaxcourt/ef-cms#6797]
As a Court User, in order to prevent a case from being scheduled for trial, I need the ability to put a "hold" on a case.

There are certain circumstances where the Court may not want a specific case to be calendared for trial.  This might happen when there is a specific, new issue that needs a test case to be decided first, or if the Court is waiting for all the cases in a group to be filed so that they can get calendared together.  In addition, while a case is waiting to be scheduled for trial, a party might file something that needs to be addressed before the case can go to trial.  As a result, Court Users need to be able to put a "hold" on a case at any time to prevent it from being scheduled for trial.

## Pre-Conditions
* Case exists in the system

## Acceptance Criteria
* Any Court User can put a "hold" on a case at any time, which would prevent that case from being scheduled for a trial session until that "hold" is removed
* Court Users could at any time view a list of cases per trial location that are in "hold" status
* List/report that is generated should list docket number and case name
* When scheduling a trial session, cases with a "hold" status would not get scheduled, even if they were "next in line" sequentially
* A Court User could at any time remove the "hold" status for a case, thereby freeing it up to be scheduled

## Notes
* Consider a case status of `General Docket - At Issue (Hold for Trial)` (Confirmed w/Jessica - status not best way to proceed.)
* From meeting with Tina and Rhonda (10/8/19) - only cases that have the "General Docket (At Issue) - Ready for Trial" need to have a "Block" option. The "block" is not a case status, but should be a flag that can accompany the status.

## Security Considerations
* Does this work make you nervous about privacy or security?
* Does this work make major changes to the system?
* Does this work implement new authentication or security controls?
* Does this work create new methods of authentication, modify existing security controls, or explicitly implement any security or privacy features?

## Tasks

- [x] 1. add the "block from trial" button onto the case information - trail information card (computed for hiding / showing) (Cody ustaxcourt/ef-cms-flexion#3294)
- [x] 2. block from trial modal (sequence for opening) (Cody ustaxcourt/ef-cms-flexion#3294)
- [x] 3. create the block case sequence which should hit a block case endpoint via proxy, show green success alert after blocking (Cody ustaxcourt/ef-cms-flexion#3294)
- [x] 4. on the case information tab, show the blocked for trial red alert and button (computeds) (Cody ustaxcourt/ef-cms-flexion#3294)
- [x] 5. work on the remove block from case modal, sequence for opening it (Cody ustaxcourt/ef-cms-flexion#3294)
- [x] 6. remove block sequence which uses the proxy to hit the unblock endpoint and green success alert (Cody ustaxcourt/ef-cms-flexion#3294)
- [x] 7. block endpoint (interactor, serverless.yml, lambda, persistence) (Cody ustaxcourt/ef-cms-flexion#3289)
- [x] 8. unblock endpoint (interactor, serverless.yml, lambda, persistence) (Cody ustaxcourt/ef-cms-flexion#3289) 
- [x] 9. add flag / methods onto case entity for setting it as blocked (Rachael ustaxcourt/ef-cms-flexion#3285)
- [x] 10. need to add a blocked cases link into the reports dropdown of the top navigaton (Rachael ustaxcourt/ef-cms-flexion#3296)
- [x] 11. create blocked cases page route (Rachael ustaxcourt/ef-cms-flexion#3296)
- [x] 12. create blocked cases page table, computed for formatting (Rachael ustaxcourt/ef-cms-flexion#3296)
- [x] 13. create blocked cases page select criteria, sequence on change (Rachael ustaxcourt/ef-cms-flexion#3296)
- [x] 14. create a "get blocked cases" endpoint (interactor, serverless.yml, lambda, persistence) (Rachael ustaxcourt/ef-cms-flexion#3293)
- [x] 15. create the blocked indicator on the case detail header with the hand-paper icon (Cody)
- [x] 16. Filter out blocked cases when setting for trial (Rachael, Cody ustaxcourt/ef-cms-flexion#3301)
- [x] 17. Elasticsearch stress testing
- [x] 18. Cleanup
  - [x] clear form when going to blocked cases report (Rachael ustaxcourt/ef-cms-flexion#3301)
  - [x] clean up UI css/styles
- [x] 19. Look into elasticsearch error on case 102-19 on dev
- [x] 20. Add Date Blocked (Rachael, Cody ustaxcourt/ef-cms-flexion#3304)
  - [x] to alert-error on Case Detail (Rachael, Cody ustaxcourt/ef-cms-flexion#3304)
  - [x] to Blocked Cases Report table (Rachael, Cody ustaxcourt/ef-cms-flexion#3304)
- [x] 21. Add Empty state message to Blocked Cases Report (Kristen ustaxcourt/ef-cms-flexion#3305)
- [x] 22. Sort blocked cases in table by docket number (Rachael ustaxcourt/ef-cms-flexion#3309)

## Definition of Done (Updated 8-28-19)
**Product Owner**
 - [x]  Acceptance criteria have been met

**UX**
 - [x] Business test scenarios to meet all acceptance criteria have been written
 - [x] Usability has been validated
 - [x] Wiki has been updated (if applicable) 
- [x] Story has been tested on a mobile device (for external users only)

**Engineering**
- [x] Automated test scripts have been written
- [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Module dependencies are up-to-date and are at the latest resolvable version (npm update)
 - [x] Errors in Sonarcloud are fixed https://sonarcloud.io/organizations/flexion-github/projects
 - [x] Lambdas include CloudWatch logging of users, inputs and outputs
 - [x] Code refactored for clarity and to remove any known technical debt 
 - [x] Deployed to the dev environment
- [ ] Deployed to the stage environment
